### PR TITLE
[fix](restore) fix Restore from __keep_on_local__ throws null pointer…

### LINKED ISSUE #26675: [Bug] Restore from `__keep_on_local__` throws null pointer exception
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

Master

### What's Wrong?

1. create database and table:

```
create database if not exists db;

drop table if exists db.cnt_table;
create table if not exists db.cnt_table (
    `id` LARGEINT NOT NULL,
    `count` LARGEINT SUM DEFAULT "0")
AGGREGATE KEY(`id`)
PARTITION BY RANGE(`id`)
(
    PARTITION p1 VALUES LESS THAN ("100"),
    PARTITION p2 VALUES LESS THAN ("200"),
    PARTITION p3 VALUES LESS THAN ("300"),
    PARTITION p4 VALUES LESS THAN ("400"),
    PARTITION p5 VALUES LESS THAN ("500"),
    PARTITION p6 VALUES LESS THAN ("600"),
    PARTITION p7 VALUES LESS THAN MAXVALUE
)
DISTRIBUTED BY HASH(`id`) BUCKETS 2
PROPERTIES (
    "replication_num" = "1"
);

INSERT INTO db.cnt_table VALUES
(1, 10),
(2, 32),
(3, 40),
(4, 33),
(4, 40);
```

2. Backup to `__keep_on_local__`.

```
backup snapshot snapshot_name TO __keep_on_local__;
```

3. Wait backup finish and restore it:

```
mysql> show backup;
+-------+---------------+--------+----------+--------------------------------+---------------------+----------------------+---------------------+--------------+-----------------+-$--------+------------+--------+---------+
| JobId | SnapshotName  | DbName | State    | BackupObjs                     | CreateTime          | SnapshotFinishedTime | UploadFinishedTime  | FinishedTime | UnfinishedTasks | $rogress | TaskErrMsg | Status | Timeout |
+-------+---------------+--------+----------+--------------------------------+---------------------+----------------------+---------------------+--------------+-----------------+-$--------+------------+--------+---------+
| 10289 | snapshot_name | db     | FINISHED | [default_cluster:db.cnt_table] | 2023-11-09 17:06:42 | 2023-11-09 17:06:47  | 2023-11-09 17:06:53 | NULL         |                 |
        |            | [OK]   | 86400   |
+-------+---------------+--------+----------+--------------------------------+---------------------+----------------------+---------------------+--------------+-----------------+-$--------+------------+--------+---------+
1 row in set (0.00 sec)

mysql> restore snapshot snapshot_name FROM __keep_on_local__
    -> ;
ERROR 1105 (HY000): Unexpected exception: null
```

### What You Expected?

Restore success.

### How to Reproduce?

In `BackupHandler::restore`, it should read job info via `BackupHandler::getSnapshot` if job info from stmt is null.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yuxuan-luo**
… exception(#26675)

## Proposed changes

Issue Number: close #26675

1. 如果 stmt 中没有携带 jobInfo，则调用 BackupHandler.getSnapshot 从内存中找，找不到报错
2. 如果有 jobInfo，则需要检查 extraInfo，也就是现在的逻辑不变

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

**Comment by JackDrogon** [comment]
Disable restore from __keep_on_local__  from sql api. if in sql the repo __keep_on_local__, just throw `wrong fro sql call， please use thrift rpc`

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — CHANGES_REQUESTED** [review]
Please add a regression test.

File: fe/fe-core/src/main/java/org/apache/doris/backup/BackupHandler.java [hunk 1]
```diff
@@ -408,6 +408,10 @@ private void backup(Repository repository, Database db, BackupStmt stmt) throws
     private void restore(Repository repository, Database db, RestoreStmt stmt) throws DdlException {
         BackupJobInfo jobInfo;
         if (stmt.isLocal()) {
+            if (stmt.getJobInfo() == null) {
```
Comment (JackDrogon) [inline fe/fe-core/src/main/java/org/apache/doris/backup/BackupHandler.java:411]: check this in Restore analyze, maybe u need to mark from sql or thrift in RestoreStmt

**Review by JackDrogon — CHANGES_REQUESTED** [review]

**Comment by yuxuan-luo** [comment]
run build all

File: regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        // Check the error message
+        assertTrue(e.message.contains("not supported"))
+    }
+
+    while (!syncer.checkAllRestoreFinish(dbName)) {
+        Thread.sleep(3000)
+    }
+
+    result = sql "SELECT * FROM ${dbName}.${tableName}"
+    assertEquals(result.size(), values.size());
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy:85]:
```suggestion
```

**Review by w41ter — CHANGES_REQUESTED** [review]

File: regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_backup_restore_keep_on_local", "backup_restore") {
+    String repoName = "__keep_on_local__"
+    String dbName = "backup_restore_db"
+    String tableName = "test_backup_restore_table"
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy:21]:
```suggestion
suite("test_backup_restore_keep_on_local", "backup_restore") {
    String repoName = "__keep_on_local__"
    String dbName = "backup_restore_keep_on_local_db"
    String tableName = "test_backup_restore_table"
```

**Review by w41ter — CHANGES_REQUESTED** [review]

**Comment by w41ter** [comment]
run buildall

**Comment by yuxuan-luo** [comment]
run build all

File: regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            )
+        """
+    } catch (Exception e) {
+        // Check the error message
+        assertTrue(e.message.contains("not supported"))
+    }
+
+    sql "DROP TABLE ${dbName}.${tableName} FORCE"
+    sql "DROP DATABASE ${dbName} FORCE"
+    sql "DROP REPOSITORY `${repoName}`"
```
Comment (w41ter) [inline regression-test/suites/backup_restore/test_backup_restore_keep_on_local.groovy:77]:
```suggestion
```

**Comment by w41ter** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**